utils/v2/connector/impl/DataPackageUtils.ts

```diff
@@ -0,0 +1,91 @@
+import EvmPriceSigner from "redstone-node/dist/src/signers/EvmPriceSigner";
+import _ from "lodash";
+import { PriceDataType, SignedPriceDataType } from "../PriceFeedConnector";
+import { SignedDataPackageResponse } from "./fetchers/Fetcher";
+import { PriceFeedOptions, ValueSelectionAlgorithm } from "./RedStonePriceFeed";
+
+const priceSigner = new EvmPriceSigner();
+
+export function selectDataPackage(
+  packages: SignedDataPackageResponse[],
+  valueSelectionAlgorithm: ValueSelectionAlgorithm,
+): SignedDataPackageResponse {
+  const sortedPackages = [...packages];
+  sortedPackages.sort((p1, p2) => p1.timestamp - p2.timestamp); // sorting prices from oldest to newest
+  switch (valueSelectionAlgorithm) {
+
+    // TODO: improve the implementation for first-valid
+    // We don't really need to wait for all responses within
+    // the first-valid method. It's enough to receive the first
+    // valid answer
+    case "first-valid":
+      return packages[packages.length - 1];
```

Comment:
Hmm, maybe I'm missing sth, but I don't see much added value with having "first-valid" vs "newest-valid" here.

Reply:
`first-valid` will work in a completely different way than `newest-valid`. It will update the data fetching logic and will not wait for all responses. Instead, it will listen on each response and will ise the first valid one. It allows to minimise time of data fetching and can be used for dapps that need to have data with the lowest possible latency.

`newest-valid` waits for all responses and selects the valid one with the biggest timestamp

Reply:
But is it implemented yet?

Reply:
No, I wanted to implement it after adding few mirror nodes. But can do it now also

Reply:
It's already implemented.